test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_work_groups.cpp

```diff
@@ -0,0 +1,300 @@
+            !clGetDeviceInfo(
+                device, CL_DEVICE_MUTABLE_DISPATCH_CAPABILITIES_KHR,
+                sizeof(mutable_capabilities), &mutable_capabilities, nullptr)
+            && mutable_capabilities
+                & CL_MUTABLE_DISPATCH_ASSERT_NO_ADDITIONAL_WORK_GROUPS_KHR;
+
+        return !no_additional_wgs_support
+            || BasicMutableCommandBufferTest::Skip();
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicMutableCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicMutableCommandBufferTest::SetUp failed");
+
+        error = SetUpKernel();
+        test_error(error, "SetUpKernel failed");
+
+        out_of_order_queue = clCreateCommandQueue(
+            context, device, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, &error);
+        test_error(error, "Unable to create command queue to test with");
+
+        cl_command_buffer_properties_khr properties[3] = {
+            CL_COMMAND_BUFFER_FLAGS_KHR, CL_COMMAND_BUFFER_MUTABLE_KHR, 0
+        };
+
+        command_buffer = clCreateCommandBufferKHR(1, &out_of_order_queue,
+                                                  properties, &error);
+        test_error(error, "clCreateCommandBufferKHR failed");
+
+        return CL_SUCCESS;
+    }
+
+    cl_int Run() override
+    {
+        const char *num_groups_kernel =
+            R"(
+                __kernel void sample_test(__global int *dst)
+            {
+                size_t tid = get_global_id(0);
+                dst[tid] = get_num_groups(0);
+            })";
+        cl_int error = create_single_kernel_helper(
+            context, &program, &kernel, 1, &num_groups_kernel, "sample_test");
+        test_error(error, "Creating kernel failed");
+
+        clMemWrapper stream = clCreateBuffer(context, CL_MEM_READ_WRITE,
+                                             sizeToAllocate, nullptr, &error);
+
+        error = clSetKernelArg(kernel, 0, sizeof(cl_mem), &stream);
+        test_error(error, "Unable to set indexed kernel arguments");
+
+        // Record an ND-range kernel of the kernel above in the command buffer
+        // with a non-null local work size so that the resulting number of
+        // workgroups will be greater than 1.
+        error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, config.ndrange_properties, kernel, 1,
+            nullptr, &global_work_size, &local_work_size, 0, nullptr, nullptr,
+            &command);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        clEventWrapper events[2];
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, &events[0]);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        clWaitForEvents(1, &events[0]);
```

Comment:
```suggestion
        error = clWaitForEvents(1, &events[0]);
```
Same for other `clWaitForEvents` calls